### ISSUE TITLE
Problem: No way to modify currencies

### ISSUE THREAD

**Issue by anbud** [Blockrazor/blockrazor#164]
Problem: Currently, once a user adds a new currency, the data is fixed and there's no way to edit it (apart from modifying it in the database)

Solution: Add a way for users to propose changes to currencies which would have to be approved by a moderator. The simplest way would be to reuse some of the UI on `addCoin` page.

**Comment by gazhayes** [comment]
That should work. The only thing is that we may need to have some kind of voting (agree + disagree) rather than just a single 'approval' to make it a bit more fuckery-tolerant.

**Comment by gazhayes** [comment]
@anbud if you haven't started on this I'd like to assign @AshHimself because I have a fair bit of Elo stuff for you to do and you are more familiar with that part of the codebase.

**Comment by AshHimself** [comment]
Ok so I propose the following;
- Add button currency page to 'Propose Change' (Anyone can propose a change)
- New view page to propose changes a long with a new collection (or use pendingcurrencies with new attributes)
- New view to track currency change requests. 
 - Who should be able to vote on currency changes? (Moderators?)
- Then we determine what approval actually means. 3 votes? Majority rules based on moderators?

**Comment by gazhayes** [comment]
This whole issue introduces another problem: the currency detail page doesn't actually have any details at present.

I was playing with using tabs or something to show more details but I don't think this is very good from a UI perspective.

On the 'home' page I'm planning on adopting a grid layout something like https://atmospherejs.com (I'll log an issue now explaining the problem). So anyway, the home page is not the place we should be proposing changes, but the currency detail page (which does not currently have much in the way of details). This issue should probably be subordinate to that.

**Comment by AshHimself** [comment]
Got it.. well I can make a start on the editing components today and then we can tweak if required as we go forward.

**Comment by AshHimself** [comment]
@gazhayes - I have created the edit form, and have got proposed currency changes going into the pendingcurrency collection stating proposedchange:true.

Doing this created a new problem, there is no easy way to see what the new changes were. e.g. bob change currency symbol from eth to etm. Thoughts? or worry about this later. I am going to submit the voting of pending changes as a seperate PR. Is that ok or you want it all together.

**Comment by gazhayes** [comment]
Each proposed change should be indepenedent, e.g. proposing to change eth to etm should be an independent item to approve than changing the ethereum premine amount. So it's probably better to use a new collection just for proposed changes and merge these into the currencies collection on approval. As well as this, users need to be able to add a note to the change (e.g. provide some evidence of appropriate).

**Comment by AshHimself** [comment]
Not a problem, i'll change the logic to reflect your changes.

**Comment by gazhayes** [comment]
Just in case it wasn't clear enough, that should solve the problem you are facing regarding previous and proposed values because you can just copy those values to the new collection.

**Comment by AshHimself** [comment]
Step ahead of you mate :)

**Comment by gazhayes** [comment]
Also remember to make use of `stash` and `pop` and keep your master brach synced with this master branch so that there aren't conflicts when you send a pull request.

**Comment by AshHimself** [comment]
I just created a PR for this. At this moment in time the proposed changes are visible but I wanted to discuss how changes should be merged back into the currencies collections. Need to discuss/agree on the flow. This took A LOT longer then I thought to do as there is so much logic between all the fields.

e.g. 3 votes then its merged or something more complex. Thoughts?

**Comment by gazhayes** [comment]
You can just send through a PR if it's a small issue, just be very clear in the commit message(s) regarding problem and solution.

**Comment by gazhayes** [comment]
> I wanted to discuss how changes should be merged back into the currencies collections
> e.g. 3 votes then its merged or something more complex. Thoughts?

I guess 3 votes should do it. We can always change it later if there's a problem.

**Comment by AshHimself** [comment]
@gazhayes - 
Ok, so three votes and its merged. I'll make this a var in our properties.js so we can change if need be. I'll do auditing on the votes as well to ensure votes are not coming in from people on multiple accounts (even though we are doing our best to avoid this).

**Comment by AshHimself** [comment]
@gazhayes - I have just created a PR for the voting functionality. We can change the voting restriction in the properties.js file. 

Once the threshold is reached the status is change from 'pending review' to 'merged'.
Once the status is merged the proposed change is merged into the currencies collection.

An additional feature to this is that we limit voting from the same IP address? This is possible as I am capturing IP on each vote within the votingMetrics object.

Let me know if you would like any changes made..

**Comment by gazhayes** [comment]
That should be OK. @anbud has introduced a feature where no one can sign up or log in with an IP address that anyone else has used.

**Comment by anbud** [comment]
@gazhayes, the current behavior for the IP tracking system is that the user only gets flagged (`flags` in `UserData` collection) if he/she used a previously used IP address. Do you want to completely disable signup/login for users using a previously used IP address? Cross-reference #39.

**Comment by gazhayes** [comment]
@anbud no that's ok for now, as long as a flag is set we can work out the rest later.

**Comment by AshHimself** [comment]
Yeh I don't think blocking by IP is a good idea. Some public places all share the same IP address externally.
